src/SFA.DAS.AANHub.Data/Repositories/CalendarEventsReadRepository.cs

```diff
@@ -1,24 +1,62 @@
-﻿using System.Diagnostics.CodeAnalysis;
-using Microsoft.EntityFrameworkCore;
+﻿using Microsoft.EntityFrameworkCore;
 using SFA.DAS.AANHub.Domain.Entities;
 using SFA.DAS.AANHub.Domain.Interfaces.Repositories;
+using SFA.DAS.AANHub.Domain.Models;
+using System.Diagnostics.CodeAnalysis;
 
 namespace SFA.DAS.AANHub.Data.Repositories;
 
 [ExcludeFromCodeCoverage]
 internal class CalendarEventsReadRepository : ICalendarEventsReadRepository
 {
-    private readonly AanDataContext _aanDataContext;
-    public CalendarEventsReadRepository(AanDataContext aanDataContext) => _aanDataContext = aanDataContext;
+	private readonly AanDataContext _aanDataContext;
+	public CalendarEventsReadRepository(AanDataContext aanDataContext) => _aanDataContext = aanDataContext;
 
-    public async Task<CalendarEvent?> GetCalendarEvent(Guid id) => 
-        await _aanDataContext
-            .CalendarEvents
-            .AsNoTracking()
-            .Where(m => m.Id == id)
-            .Include(x => x.Attendees.Where(a => a.IsActive))
-            .ThenInclude(x => x.Member)
-            .Include(x => x.EventGuests)
-            .Include(x => x.Calendar)
-            .SingleOrDefaultAsync();
+	public async Task<CalendarEvent?> GetCalendarEvent(Guid id) =>
+		await _aanDataContext
+			.CalendarEvents
+			.AsNoTracking()
+			.Where(m => m.Id == id)
+			.Include(x => x.Attendees.Where(a => a.IsActive))
+			.ThenInclude(x => x.Member)
+			.Include(x => x.EventGuests)
+			.Include(x => x.Calendar)
+			.SingleOrDefaultAsync();
+
+	public async Task<List<CalendarEventSummary>> GetCalendarEvents(Guid memberId, CancellationToken cancellationToken)
+	{
+		FormattableString sql = $@"select	
+                            CE.Id as CalendarEventId, 
+	                        C.CalendarName,
+	                        CE.EventFormat, 
+	                        CE.StartDate as [Start], 
+	                        CE.EndDate as [End],
+	                        CE.[Description], 
+	                        CE.Summary,
+	                        CE.[Location],   
+	                        CE.Postcode,
+	                        CE.Latitude,
+	                        CE.Longitude,
+	                        CASE  WHEN (MPLat.ProfileValue is null OR ltrim(MPLat.ProfileValue) = '') THEN null
```

Comment:
We add ProfileValues and it is decided that there should not be any empty values in profile. So only null check should be sufficient.

Reply:
ok